app/controllers/api/v1/display_ads_controller.rb

```diff
@@ -15,11 +15,40 @@ def show
         render json: @display_ad
       end
 
+      def create
+        @display_ad = DisplayAd.new(permitted_params)
+        result = @display_ad.save
+        render json: @display_ad, status: (result ? :ok : :unprocessable_entity)
+      rescue ArgumentError => e
+        # enums raise ArgumentError exceptions on unexpected inputs!
+        render json: { error: e }, status: :unprocessable_entity
+      end
+
+      def update
+        @display_ad = DisplayAd.find(params[:id])
+        result = @display_ad.update(permitted_params)
+        render json: @display_ad, status: (result ? :ok : :unprocessable_entity)
+      rescue ArgumentError => e
+        # enums raise ArgumentError exceptions on unexpected inputs!
+        render json: { error: e }, status: :unprocessable_entity
+      end
+
+      def destroy
+        @display_ad = DisplayAd.find(params[:id])
+        result = @display_ad.update(published: false)
+        render json: @display_ad, status: (result ? :ok : :unprocessable_entity)
+      end
+
       private
 
       def require_admin
         authorize DisplayAd, :access?, policy_class: InternalPolicy
       end
+
+      def permitted_params
+        params.permit :approved, :body_markdown, :display_to, :name,
```

Comment:
Should we handle :tags via the API as well?

Reply:
Yes, it should, good catch!